src/client/components/pages/Courses/tableFields.tsx

```diff
@@ -175,21 +181,25 @@ export const formatMeetings = (
           ))}
         </TableCellList>
         <BorderlessButton
-          id={`${courseId}-${sem}-edit-meetings-button`}
-          onClick={() => coursesPageContext
-            && coursesPageContext.onMeetingEdit({
-              course,
-              term: sem,
-            })}
+          id={`${courseId}-${term}-edit-meetings-button`}
+          onClick={() => { setModalVisible(true); }}
           variant={VARIANT.INFO}
-          forwardRef={coursesPageContext?.currentCourseInstance?.course.id
-            === courseId
-            && coursesPageContext.currentCourseInstance.term === sem
-            ? coursesPageContext.meetingEditButtonRef
-            : null}
+          forwardRef={buttonRef}
         >
           <FontAwesomeIcon icon={faEdit} />
         </BorderlessButton>
+        {modalVisible && (
+          <MeetingModal
+            isVisible={modalVisible}
```

Comment:
Ah, yes it is. I had done that because there were a ton of errors showing in the console because the the modal tries to focus itself when it first renders, regardless of whether the `Modal` itself is visible. I set up a different workaround in 0cab5d0e. 

On that subject, that "focus on opening" behavior is something that could probably be baked into the Modal itself, so we don't have to re-implement it every time. I'll look into that and maybe open a ticket on Mark One.

Reply:
Interesting edge-case. That makes sense though.